gate-web/src/main/groovy/com/netflix/spinnaker/gate/controllers/ApplicationController.groovy

```diff
@@ -54,8 +55,17 @@ class ApplicationController {
 
   @ApiOperation(value = "Retrieve a list of applications")
   @RequestMapping(method = RequestMethod.GET)
-  List<Map> getAllApplications() {
-    applicationService.getAllApplications()
+  List<Map> getAllApplications(
+    @ApiParam(name = "account", required = false, value = "filters results to only include applications deployed in the specified account")
+    @RequestParam(value = "account", required = false) String account) {
+    List<Map> allApplications = applicationService.getAllApplications()
+    if (account) {
+      return allApplications.findAll { app ->
+        String[] appAccounts = ((String) app.accounts ?: "").split(",")
+        appAccounts.contains(account)
```

Comment:
maybe `toLowerCase()` some stuff here to protect against someone accidentally passing in a non-lower case parameter (accounts on the application _should_ be lowercase already but you might want to make that explicit as well).